### PR TITLE
Fix Twitter API Oauth Error

### PR DESCRIPTION
**Pull request by dblythy**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Issue Description

Fixes the failing test explained in #7369.

### Approach

Changes the endpoint that the GET request makes from https://api.twitter.com/1.1/help/configuration.json to https://api.twitter.com/1.1/oauth/request_token.

### CODE REVIEW

**Comment by mtrezza** [comment]
Thanks for opening this PR!

That is interesting, did you find any reference in the Twitter docs why this has changes?

**Comment by dblythy** [comment]
Hey @mtrezza, thanks for your comments. You’re correct, as the current endpoint has worked for so long, perhaps it’s a Twitter issue. The change in this PR simply changes the endpoint without attempting to understand why the existing endpoint no longer works (I’m assuming it is depreciated). Perhaps the best approach is wait until the endpoint works again?

**Comment by mtrezza** [comment]
This still doesn't make any sense to me. This PR changes to a completely different endpoint.

**Comment by mtrezza** [comment]
After identifying the [underlying issue](https://github.com/parse-community/parse-server/issues/7369#issuecomment-826159851), I suggest the following changes:

- Use a different endpoint
  The suggested endpoint `/1.1/oauth/request_token` is not a valid endpoint according to the docs. It returns the expected invalid auth response but so does `/1.1/oauth/blabla`. Also, it is a documented as a POST endpoint, but the test is for a GET endpoint. It's surprising that the test passes, likely because the Twitter API is quite graceful. Also, the endpoint does not require auth, it is an endpoint to *authenticate*, not to request a resource that *requires authentication*, which is what the test is about.

- Use endpoint `/1.1/account/settings.json`
  It's the same endpoint the `POST` test uses below the `GET` test; the endpoint is actually a POST and GET endpoint, so we don't need to rely on two dummy endpoints.

- Add a comment to the test
  To avoid confusion in the future, we should add a line to both tests (POST and GET) that these endpoints are just picked as dummies to test auth failing, for example:
  > Note: This endpoint has been chosen to make a request to an endpoint that requires OAuth which failed due to missing authentication. Any other endpoint from the [Twitter API](https://developer.twitter.com/en/docs/api-reference-index) that requires OAuth can be used instead in case the currently used endpoint deprecates.

**Comment by mtrezza** [comment]
After identifying the [underlying issue](https://github.com/parse-community/parse-server/issues/7369#issuecomment-826159851), I suggest the following changes:

- Use a different endpoint
  The suggested endpoint `/1.1/oauth/request_token` is not a valid endpoint according to the docs. It returns the expected invalid auth response but so does `/1.1/oauth/blabla`. Also, it is a documented as a POST endpoint, but the test is for a GET endpoint. It's surprising that the test passes, likely because the Twitter API is quite graceful. Also, the endpoint does not require auth, it is an endpoint to *authenticate*, not to request a resource that *requires authentication*, which is what the test is about.

- Use endpoint `/1.1/account/settings.json`
  It's the same endpoint the `POST` test uses below the `GET` test; the endpoint is actually a POST and GET endpoint, so we don't need to rely on two dummy endpoints.

- Add a comment to the test
  To avoid confusion in the future, we should add a line to both tests (POST and GET) that these endpoints are just picked as dummies to test auth failing, for example:
  > Note: This endpoint has been chosen to make a request to an endpoint that requires OAuth which failed due to missing authentication. Any other endpoint from the [Twitter API](https://developer.twitter.com/en/docs/api-reference-index) that requires OAuth can be used instead in case the currently used endpoint deprecates.

**Comment by dblythy** [comment]
Thanks for the suggestions! I'll add the changes.

File: spec/OAuth1.spec.js
Comment (mtrezza) [inline spec/OAuth1.spec.js:96]: This endpoint is invalid, change to same as POST endpoint.

File: spec/OAuth1.spec.js
Comment (mtrezza) [inline spec/OAuth1.spec.js:96]: Add comment why this endpoint is used.

File: spec/OAuth1.spec.js
Comment (mtrezza) [inline spec/OAuth1.spec.js:94]: Change credentials to `invalid`.

**Review by mtrezza — CHANGES_REQUESTED** [review]
Formally requesting the changes so we don't merge with invalid endpoint by mistake.

File: spec/OAuth1.spec.js
Comment (mtrezza) [inline spec/OAuth1.spec.js:90]: Looks already much better. Can you just add `for a resource that requires OAuth` to both tests? To explicitly describe the purpose of the test for future changes.

**Review by mtrezza — CHANGES_REQUESTED** [review]

**Comment by mtrezza** [comment]
This looks good! Any thing you want to add before merging?

**Comment by dblythy** [comment]
Nope! I'm ready to merge 😊

**Review by mtrezza — COMMENTED** [review]
LGTM Thanks for tackling this so quickly!

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.0.0-beta.1](https://github.com/parse-community/parse-server/releases/tag/5.0.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.0.0](https://github.com/parse-community/parse-server/releases/tag/5.0.0)

**Pull request merged**
